hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/realtime/AbstractRealtimeRecordReader.java

```diff
@@ -323,13 +340,21 @@ private static Schema addPartitionFields(Schema schema, List<String> partitionin
    * the base split was written.
    */
   private void init() throws IOException {
-    writerSchema = new AvroSchemaConverter().convert(baseFileSchema);
-    List<String> fieldNames = writerSchema.getFields().stream().map(Field::name).collect(Collectors.toList());
+    Schema schemaFromLogFile = null;
+    HoodieTableMetaClient metaClient = new HoodieTableMetaClient(jobConf, split.getBasePath());
+    HoodieTimeline completedTimeline = metaClient.getActiveTimeline().getCommitsTimeline()
+        .filterCompletedInstants();
+    Collections.reverse(split.getDeltaFilePaths());
```

Comment:
Is this safe even for cases where there are log files before and after pending compaction ?

Reply:
It is since we get it in the reverse order from HoodieRealtimeInputFormat class which calls the fileslice.getSplits() which in turn has sorted it correctly. But that's too much context to keep and can mess up anytime, so sorting it again here anyways.